sqlalchemy_utils/types/enriched_datetime/enriched_datetime_type.py

```diff
@@ -0,0 +1,80 @@
+from sqlalchemy import types
+
+from ...exceptions import ImproperlyConfigured
+from ..scalar_coercible import ScalarCoercible
+from .arrow_datetime import ArrowDatetime
+from .pendulum_datetime import PendulumDatetime
+
+arrow = None
+try:
+    import arrow
+except ImportError:
+    pass
+
+pendulum = None
+try:
+    import pendulum
+except ImportError:
+    pass
+
+
+class EnrichedDateTimeType(types.TypeDecorator, ScalarCoercible):
+    """
+    Supported for arrow and pendulum.
+
+    Example::
+
+
+        from sqlalchemy_utils import EnrichedDateTimeType
+        import pendulum
+
+
+        class User(Base):
+            __tablename__ = 'user'
+            id = sa.Column(sa.Integer, primary_key=True)
+            created_at = sa.Column(EnrichedDateTimeType(type="pendulum"))
+            # created_at = sa.Column(EnrichedDateTimeType(type="arrow"))
+
+
+        user = User()
+        user.created_at = pendulum.now()
+        session.add(user)
+        session.commit()
+    """
+    impl = types.DateTime
+
+    def __init__(self, type="pendulum", *args, **kwargs):
```

Comment:
I think a better way would be to pass the ``date_class`` as an argument.

```
def __init__(self, date_class= PendulumDate, *args, **kwargs):
```

Reply:
The constructor of those custom date classes could throw an error if the given package is not installed.